src/ui/layout/constraints.md

```diff
@@ -118,7 +119,53 @@ Flutter’s layout engine has a few important limitations:
   the parent doesn't have enough information to align it,
   then the child's size might be ignored.
   **Be specific when defining alignment.**
+
+{{site.alert.note}}
+  In Flutter, widgets are rendered by their underlying
+  [`RenderBox`][] objects.
+  Many boxes in Flutter,
+  especially those that just take a single child,
+  pass their constraint on to their children.
+  This means that if you nest a bunch of boxes inside each other
+  at the root of your application's render tree,
+  they'll all exactly fit in each other,
+  forced by these tight constraints.
```

Comment:
this is the first mention of "tight" and i'm not sure what the antecedent for "these tight constraints" is.